src/Illuminate/View/Concerns/ManagesLoops.php

```diff
@@ -22,7 +23,13 @@
      */
     public function addLoop($data)
     {
-        $length = is_array($data) || $data instanceof Countable ? count($data) : null;
+        $length = null;
+
+        if (is_array($data) || $data instanceof Countable) {
+            $length = count($data);
+        } elseif ($data instanceof Traversable) {
+            $length = iterator_count($data);
```

Comment:
This breaks when the data used in the view comes from a generator (using `yield`)...
```
Exception(code: 0): Cannot traverse an already closed generator
```
@halloei @taylorotwell